fpga/common.mk

```diff
@@ -0,0 +1,119 @@
+# See LICENSE for license details.
+
+# Required variables:
+# - MODEL
+# - PROJECT
+# - CONFIG_PROJECT
+# - CONFIG
+# - BUILD_DIR
+# - FPGA_DIR
+
+# Optional variables:
+# - EXTRA_FPGA_VSRCS
+
+# export to bootloader
+export ROMCONF=$(BUILD_DIR)/$(CONFIG_PROJECT).$(CONFIG).rom.conf
+
+# export to fpga-shells
+export FPGA_TOP_SYSTEM=$(MODEL)
+export FPGA_BUILD_DIR=$(BUILD_DIR)/$(FPGA_TOP_SYSTEM)
+export fpga_common_script_dir=$(FPGA_DIR)/common/tcl
+export fpga_board_script_dir=$(FPGA_DIR)/$(BOARD)/tcl
+
+export BUILD_DIR
+
+EXTRA_FPGA_VSRCS ?=
+PATCHVERILOG ?= ""
+BOOTROM_DIR ?= ""
+
+base_dir=$(abspath ..)
+export rocketchip_dir := $(base_dir)/generators/rocket-chip
+SBT ?= java -jar $(rocketchip_dir)/sbt-launch.jar ++2.12.10
```

Comment:
I think we want to dedupe these variables as much as possible to integrate with the chipyard make system.

Reply:
I think I solved this with #669